src/main/cpp/src/decimal_utils.cu

```diff
@@ -968,6 +974,7 @@ namespace cudf::jni {
 std::unique_ptr<cudf::table> multiply_decimal128(cudf::column_view const& a,
                                                  cudf::column_view const& b,
                                                  int32_t product_scale,
+                                                 bool const& cast_interim_result,
```

Comment:
```suggestion
                                                 bool const cast_interim_result,
```
No real advantage to passing it as a pointer if it is const and smaller than a pointer.